pkgs/development/compilers/haxe/default.nix

```diff
@@ -1,14 +1,39 @@
-{ lib, stdenv, fetchgit, coreutils, ocamlPackages, zlib, pcre, neko }:
-
-let inherit (ocamlPackages) ocaml camlp4; in
+{ lib, stdenv, fetchgit, coreutils, ocaml-ng, zlib, pcre, neko, mbedtls }:
 
 let
-  generic = { version, sha256, prePatch }:
+  ocamlDependencies = version:
+    if lib.versionAtLeast version "4.0"
+    then with ocaml-ng.ocamlPackages; [
+      ocaml
+      findlib
+      sedlex_2
+      xml-light
+      xml-light
+      ptmap
+      camlp5
+      sha
+      dune_2
+      luv
```

Comment:
```suggestion
      camlp5
      dune_2
      findlib
      luv
      ptmap
      sedlex_2
      sha
      xml-light
```

Reply:
Alphabetical sorting doesn't matter.

Reply:
why is `xml-light` duplicated in the list?

Reply:
Oversight. Thanks.

Reply:
Where does the sorting come from?

Reply:
There is none and sorting is also generally not a good idea: If it were to make sense, we would need to enforce it. Enforcing sorted argument / input lists in nixpkgs would lead to more merge conflicts than the trouble would be worth.

Even in files which are supposed to be alphabetically sorted, like `all-packages.nix` etc. this rule is not enforced because it is not feasible.